radial_build_tools.py

```diff
@@ -56,6 +54,37 @@ def __init__(self, build, **kwargs):
         ):
             self.__setattr__(name, kwargs[name])
 
+        # Initialize colors for each layer
+        self.colors = self.assign_colors()
+
+    def assign_colors(self):
+        """
+        Assign colors to each layer. If a color is provided, use it.
+        If not, generate a random color that hasn't been used yet.
+        """
+        used_colors = set()
```

Comment:
I think we want to get all of the used colors before looping - small chance to randomly get the same color as a user assigned but I think technically possible with how this is written

Reply:
`used_colors` could be a class variable that gets set on initialization. Same with `available_colors`, could use a set for both then, and remove colors from `available_colors` and add them to `used_colors` as they get used